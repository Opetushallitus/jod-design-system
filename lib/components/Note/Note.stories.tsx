import type { StoryObj } from '@storybook/react-vite';
import { fn } from 'storybook/test';
import type { TitledMeta } from '../../utils';

import { Note, NoteStack, NoteStackProvider, useNoteStack } from '.';
import { JodOpenInNew } from '../../icons';
import { Button } from '../Button/Button';
import { NoteProps } from './Note';

const meta = {
  title: 'Popups/Note',
  component: Note,
  tags: ['autodocs'],
  parameters: {
    layout: 'fullscreen',
    controls: {
      exclude: ['readMoreComponent', 'onCloseClick'],
    },
  },
  globals: {
    viewport: { value: 'fullwidth' },
  },
} satisfies TitledMeta<typeof Note>;

export default meta;

type Story = StoryObj<typeof meta>;

const title = 'Lorem ipsum';
const description = 'Ipsum dolor est nonummy sit amet accusata mediocre me tulsa opus.';

// Add some bottom padding to stories so that the "Show code" button doesn't cover the content
const decorators: Story['decorators'] = [
  (Story) => (
    <div className="ds:pb-[25px]">
      <Story />
    </div>
  ),
];

const readMoreComponent = <Button variant="plain" label="Lue lisää" className="ds:text-primary-gray!" />;

export const ConfirmationNote: Story = {
  decorators,
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/design/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=14337-69983',
    },
    docs: {
      description: {
        story: 'This is a success note component for displaying a text.',
      },
    },
  },
  args: {
    title,
    description,
    readMoreComponent,
    variant: 'success',
    onCloseClick: fn(),
  },
};

export const LongTitleText: Story = {
  decorators,
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/design/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=14337-69983',
    },
    docs: {
      description: {
        story: 'This is a success note component for displaying a long title text.',
      },
    },
  },
  args: {
    title: 'Lorem ipsum dolor est nonummy noblem ester!',
    description,
    readMoreComponent,
    variant: 'success',
    onCloseClick: fn(),
  },
};

export const PermanentNote: Story = {
  decorators,
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/design/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=14337-70141',
    },
    docs: {
      description: {
        story: 'This is a error note component for displaying a permanent note.',
      },
    },
  },
  args: {
    title,
    description,
    variant: 'error',
    permanent: true,
  },
};

export const NoCTA: Story = {
  decorators,
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/design/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=14337-70049',
    },
    docs: {
      description: {
        story: 'This is a success note component without a call to action.',
      },
    },
  },
  args: {
    title,
    description,
    onCloseClick: fn(),
  },
};

export const WarningNote: Story = {
  decorators,
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/design/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=14337-70005',
    },
    docs: {
      description: {
        story: 'This is a warning note component for displaying a text.',
      },
    },
  },
  args: {
    title,
    description,
    variant: 'warning',
  },
};

export const ErrorNote: Story = {
  decorators,
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/design/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=2217-8777&t=XinajcpkjOyuElVJ-4',
    },
    docs: {
      description: {
        story: 'This is a warning note component for displaying a text.',
      },
    },
  },
  args: {
    title,
    description,
    variant: 'error',
  },
};

export const feedback: Story = {
  decorators,
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/design/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=14337-70080',
    },
    docs: {
      description: {
        story: 'This is a feedback note component for displaying a text.',
      },
    },
  },
  args: {
    title: 'Kokeile palvelua',
    description: 'Kerro mielipiteesi ja auta meitä kehittämään palvelua.',
    variant: 'feedback',
    readMoreComponent: (
      <Button variant="white" label="Siirry kyselyyn" icon={<JodOpenInNew />} iconSide="right" size="sm" />
    ),
    onCloseClick: fn(),
  },
};

export const WithNoteStack: Story = {
  decorators: [
    (Story) => (
      <div className="ds:h-[400px] ds:overflow-y-auto">
        <Story />
      </div>
    ),
  ],
  render: (args) => {
    const NoteStackDemo = () => {
      const { addNote } = useNoteStack();
      const variants = ['success', 'error', 'warning', 'feedback'] as NoteProps['variant'][];
      return (
        <div className="ds:flex ds:flex-col ds:gap-5 ds:relative">
          <div className="ds:fixed ds:top-0 ds:w-full">
            <NoteStack showAllText="Näytä kaikki" />
          </div>
          <div className="ds:mt-[200px] ds:ml-3 ds:flex ds:gap-5">
            {variants.map((variant) => (
              <Button
                key={variant}
                variant="accent"
                size="sm"
                label={`Add ${variant} note`}
                onClick={() => addNote({ ...args, variant, title: variant?.toLocaleUpperCase() })}
              />
            ))}
            <Button
              variant="red-delete"
              size="sm"
              label={`Add permanent note`}
              onClick={() => addNote({ ...args, variant: 'error', title: 'PERMANENT', permanent: true })}
            />
          </div>
        </div>
      );
    };

    return (
      <NoteStackProvider>
        <NoteStackDemo />
      </NoteStackProvider>
    );
  },
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/design/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=14337-67043',
    },
    docs: {
      description: {
        story:
          'Notes inside a stack component. The stack uses a hook to manage the notes. Notes are sorted by variant and permanent status. When user scrolls the page down, all notes besides permanent ones will collapse. They re-appear if user scrolls to the top of the page. You can add notes by using the provided buttons.',
      },
    },
  },
  args: {
    description,
  },
};
