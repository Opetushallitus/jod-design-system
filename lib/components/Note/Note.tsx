import { cx } from '../../cva';
import { useMediaQueries } from '../../hooks/useMediaQueries';

export interface NoteProps {
  /** Title shown on the note */
  title?: string;
  /** Description shown on the note */
  description?: string;
  /** Icon shown on the note */
  variant?: 'success' | 'warning' | 'error';
  /** Callback fired on tap/click of the close button */
  onCloseClick?: () => void;
  /** Text for read more link if variant is success */
  readMoreText?: string;
  /** Link to read more if variant is success */
  readMoreHref?: string;
}

export const Note = ({
  title,
  description,
  variant = 'success',
  onCloseClick,
  readMoreText,
  readMoreHref,
}: NoteProps) => {
  const { sm } = useMediaQueries();
  const hasReadMore = variant === 'success' && readMoreText && readMoreHref;
  return (
    <div
      role="alert"
      aria-live="assertive"
      aria-atomic="true"
      className={cx('text-black', {
        'bg-success': variant === 'success',
        'bg-warning': variant === 'warning',
        'bg-alert': variant === 'error',
      })}
    >
      <div className="mx-auto flex min-h-[42px] items-center px-5 justify-start sm:min-h-11 sm:pl-7 sm:pr-0 sm:justify-center sm:gap-x-[20%]">
        <div className="flex flex-col flex-wrap items-start py-4 sm:flex-row sm:items-center sm:gap-x-5 sm:gap-y-2 sm:py-5">
          <div className="font-poppins text-heading-4 sm:text-heading-3">{title}</div>
          <div className="mt-1 sm:mt-0 text-body-sm">{description}</div>
          {hasReadMore && !sm && (
            <a
              className="font-poppins mt-4 text-nowrap rounded-[30px] bg-white px-6 py-[10px] text-button-md hover:underline focus-visible:outline focus-visible:outline-[3px] focus-visible:outline-offset-[1.5px] focus-visible:outline-white active:bg-accent active:text-white active:no-underline"
              href={readMoreHref}
              role="button"
            >
              {readMoreText}
            </a>
          )}
        </div>
        <div className="flex items-center">
          {hasReadMore && sm && (
            <a
              className="font-poppins mx-7 text-nowrap rounded-[30px] bg-white px-6 py-[10px] text-button-md hover:underline focus-visible:outline focus-visible:outline-[3px] focus-visible:outline-offset-[1.5px] focus-visible:outline-white active:bg-accent active:text-white active:no-underline"
              href={readMoreHref}
              role="button"
            >
              {readMoreText}
            </a>
          )}
          {onCloseClick && (
            <button
              className={`ml-3 mr-0 flex sm:mr-5 ${hasReadMore ? 'sm:ml-0' : 'sm:ml-5'}`}
              type="button"
              aria-label="Close"
              onClick={onCloseClick}
            >
              <span className="material-symbols-outlined size-32 select-none">cancel</span>
            </button>
          )}
        </div>
      </div>
    </div>
  );
};
