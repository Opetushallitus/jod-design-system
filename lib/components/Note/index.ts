export { NoteStackContext } from './hooks/NoteStackContext';
export { NoteStackProvider } from './hooks/NoteStackProvider';
export { useNoteStack } from './hooks/useNoteStack';
export { Note } from './Note';
export { NoteStack, type NoteStackProps } from './NoteStack';
