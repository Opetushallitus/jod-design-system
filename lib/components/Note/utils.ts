import { NoteProps } from './Note';

export interface NoteStackNote extends NoteProps {
  id: string;
  collapsed?: boolean;
}

export const DEFAULT_MAX_NOTES = 3;
