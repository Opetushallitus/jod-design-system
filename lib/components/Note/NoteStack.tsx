import React from 'react';
import { cx } from '../../cva';
import { useCollapseOnScroll } from '../../hooks/useCollapseOnScroll';
import { getAccentBgClassForService, type ServiceVariant, tidyClasses } from '../../utils';
import { Note } from './Note';
import { useNoteStack } from './hooks/useNoteStack';
import { DEFAULT_MAX_NOTES, type NoteStackNote } from './utils';

type ServiceBarProps =
  | {
      /** Should the service bar be displayed under the navigation bar */
      showServiceBar: true;
      /** Service variant for styling */
      serviceBarVariant: ServiceVariant;
      /** Service bar title */
      serviceBarTitle?: string;
      /** Component to display on the right side of the service bar */
      serviceBarContent?: React.ReactNode;
    }
  | {
      showServiceBar?: false;
      serviceBarVariant?: never;
      serviceBarTitle?: never;
      serviceBarContent?: never;
    };
export type NoteStackProps = {
  showAllText: string;
  onShowAllClick?: () => void;
  showServiceBar?: boolean;
  serviceBarTitle?: string;
  serviceBarContent?: React.ReactNode;
  serviceVariant?: ServiceVariant;
} & ServiceBarProps;

export const NoteStack = ({
  showAllText,
  onShowAllClick,
  showServiceBar,
  serviceBarTitle,
  serviceBarContent,
  serviceVariant,
}: NoteStackProps) => {
  const { notes, setNotes, removeNote, uncollapseAll, maxNotes = DEFAULT_MAX_NOTES } = useNoteStack();
  const [serviceBarCollapsed, setServiceBarCollapsed] = React.useState(false);
  const allCollapsed = notes.every((n) => n.collapsed);
  const allUncollapsed = notes.every((n) => !n.collapsed);
  const hasUncollapsed = notes.some((n) => !n.collapsed);
  const buttonVisible = notes.length > 1 && hasUncollapsed && !allCollapsed && !allUncollapsed;

  // Initialization: collapse all but the first note
  React.useEffect(() => {
    setNotes((prev) => prev.map((n, index) => ({ ...n, collapsed: index > 0 })));
  }, [setNotes]);

  // Collapse all non-permanent notes on scroll
  const collapseMapper = (collapsed: boolean) => (notes: NoteStackNote[]) =>
    notes.map((n) => (n.permanent ? { ...n, collapsed: false } : { ...n, collapsed }));

  const onCollapse = React.useCallback(() => {
    setNotes(collapseMapper(true));
    setServiceBarCollapsed(true);
  }, [setNotes]);

  const onUncollapse = React.useCallback(() => {
    setNotes(notes.map((n, index) => (n.permanent ? { ...n, collapsed: false } : { ...n, collapsed: index > 0 })));
    setServiceBarCollapsed(false);
  }, [notes, setNotes]);

  const { resetCollapseState } = useCollapseOnScroll({
    onCollapse,
    onUncollapse,
    startupDelayMs: 500,
  });

  const getButtonColors = React.useCallback(() => {
    // If any note is permanent, use error variant, otherwise use the first note's variant
    const variant = notes.at(0)?.variant;

    return !variant
      ? ''
      : cx({
          'ds:bg-alert ds:text-white': variant === 'error',
          'ds:bg-warning ds:text-primary-gray': variant === 'warning',
          'ds:bg-success ds:text-primary-gray': variant === 'success',
          'ds:bg-secondary-3 ds:text-primary-gray': variant === 'feedback',
        });
  }, [notes]);

  const serviceBarContents = (
    <>
      {serviceBarTitle ? <span>{serviceBarTitle}</span> : null}
      {serviceBarContent ? <div>{serviceBarContent}</div> : null}
    </>
  );

  return (
    <div className="ds:flex ds:flex-col ds:absolute ds:w-full">
      {showServiceBar && (
        <div
          className={tidyClasses([
            'ds:w-full',
            'ds:flex',
            'ds:text-white',
            'ds:text-[12px]',
            'ds:sm:text-[14px]',
            'ds:transition-all',
            'ds:duration-300',
            'ds:h-8',
            serviceBarCollapsed ? 'ds:-translate-y-full ds:mt-2' : 'ds:translate-y-0',
            getAccentBgClassForService(serviceVariant || 'yksilo'),
          ])}
          data-testid="servicebar"
        >
          <div className="ds:flex ds:xl:container ds:mx-auto ds:items-center ds:justify-between ds:w-full ds:sm:px-9 ds:px-5">
            {serviceBarContents}
          </div>
        </div>
      )}

      {notes.slice(0, maxNotes).map((note) => (
        <Note
          {...note}
          key={note.id}
          className={serviceBarCollapsed ? 'ds:-translate-y-8' : ''}
          onCloseClick={() => {
            note.onCloseClick?.();
            removeNote(note.id);
          }}
        />
      ))}
      {notes.length > 1 && (
        <button
          aria-hidden={!buttonVisible}
          tabIndex={buttonVisible ? 0 : -1}
          className={tidyClasses([
            'ds:cursor-pointer',
            'ds:ml-auto',
            'ds:mr-6',
            'ds:text-primary-gray',
            'ds:text-button-sm',
            'ds:px-6',
            'ds:rounded-b-md',
            'ds:overflow-clip',
            'ds:absolute',
            'ds:py-3',
            'ds:top-full',
            'ds:right-0',
            'ds:transition-transform',
            'ds:duration-300',
            buttonVisible ? 'ds:scale-y-100' : 'ds:scale-y-0',
            serviceBarCollapsed ? 'ds:-translate-y-8' : '',
            'ds:origin-top',
            getButtonColors(),
          ])}
          onClick={() => {
            resetCollapseState();
            uncollapseAll();
            onShowAllClick?.();
          }}
        >
          <span aria-hidden={!buttonVisible}>{`${showAllText} (${notes.length})`}</span>
        </button>
      )}
    </div>
  );
};
