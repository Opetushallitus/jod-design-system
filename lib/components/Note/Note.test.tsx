import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import { Note } from './Note';

describe('Note component', () => {
  it('renders with default variant', () => {
    const { container } = render(<Note title="Test Title" description="Test Description" />);
    const noteElement = screen.getByRole('alert');
    expect(noteElement).toBeInTheDocument();
    expect(noteElement).toHaveClass('bg-success');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with success variant', () => {
    const { container } = render(<Note title="Test Title" description="Test Description" variant="success" />);
    const noteElement = screen.getByRole('alert');
    expect(noteElement).toBeInTheDocument();
    expect(noteElement).toHaveClass('bg-success');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with warning variant', () => {
    const { container } = render(<Note title="Test Title" description="Test Description" variant="warning" />);
    const noteElement = screen.getByRole('alert');
    expect(noteElement).toBeInTheDocument();
    expect(noteElement).toHaveClass('bg-warning');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with error variant', () => {
    const { container } = render(<Note title="Test Title" description="Test Description" variant="error" />);
    const noteElement = screen.getByRole('alert');
    expect(noteElement).toBeInTheDocument();
    expect(noteElement).toHaveClass('bg-alert');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with read more link', () => {
    const { container } = render(
      <Note
        title="Test Title"
        description="Test Description"
        variant="success"
        readMoreText="Read more"
        readMoreHref="/"
      />,
    );
    const readMoreLink = screen.getByRole('button', { name: 'Read more' });
    expect(readMoreLink).toBeInTheDocument();
    expect(readMoreLink).toHaveAttribute('href', '/');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with close button', () => {
    const onCloseClick = vi.fn();
    const { container } = render(
      <Note title="Test Title" description="Test Description" variant="success" onCloseClick={onCloseClick} />,
    );
    const closeButton = screen.getByLabelText('Close');
    expect(closeButton).toBeInTheDocument();
    expect(closeButton.firstChild).toHaveClass('material-symbols-outlined');
    expect(closeButton).toHaveAttribute('aria-label', 'Close');
    closeButton.click();
    expect(onCloseClick).toHaveBeenCalled();
    expect(container.firstChild).toMatchSnapshot();
  });
});
