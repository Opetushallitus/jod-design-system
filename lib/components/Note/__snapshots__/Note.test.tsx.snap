// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Note component > renders with close button 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  class="text-black bg-success"
  role="alert"
>
  <div
    class="mx-auto flex min-h-[42px] items-center px-5 justify-start sm:min-h-11 sm:pl-7 sm:pr-0 sm:justify-center sm:gap-x-[20%]"
  >
    <div
      class="flex flex-col flex-wrap items-start py-4 sm:flex-row sm:items-center sm:gap-x-5 sm:gap-y-2 sm:py-5"
    >
      <div
        class="font-poppins text-heading-4 sm:text-heading-3"
      >
        Test Title
      </div>
      <div
        class="mt-1 sm:mt-0 text-body-sm"
      >
        Test Description
      </div>
    </div>
    <div
      class="flex items-center"
    >
      <button
        aria-label="Close"
        class="ml-3 mr-0 flex sm:mr-5 sm:ml-5"
        type="button"
      >
        <span
          class="material-symbols-outlined size-32 select-none"
        >
          cancel
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Note component > renders with default variant 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  class="text-black bg-success"
  role="alert"
>
  <div
    class="mx-auto flex min-h-[42px] items-center px-5 justify-start sm:min-h-11 sm:pl-7 sm:pr-0 sm:justify-center sm:gap-x-[20%]"
  >
    <div
      class="flex flex-col flex-wrap items-start py-4 sm:flex-row sm:items-center sm:gap-x-5 sm:gap-y-2 sm:py-5"
    >
      <div
        class="font-poppins text-heading-4 sm:text-heading-3"
      >
        Test Title
      </div>
      <div
        class="mt-1 sm:mt-0 text-body-sm"
      >
        Test Description
      </div>
    </div>
    <div
      class="flex items-center"
    />
  </div>
</div>
`;

exports[`Note component > renders with error variant 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  class="text-black bg-alert"
  role="alert"
>
  <div
    class="mx-auto flex min-h-[42px] items-center px-5 justify-start sm:min-h-11 sm:pl-7 sm:pr-0 sm:justify-center sm:gap-x-[20%]"
  >
    <div
      class="flex flex-col flex-wrap items-start py-4 sm:flex-row sm:items-center sm:gap-x-5 sm:gap-y-2 sm:py-5"
    >
      <div
        class="font-poppins text-heading-4 sm:text-heading-3"
      >
        Test Title
      </div>
      <div
        class="mt-1 sm:mt-0 text-body-sm"
      >
        Test Description
      </div>
    </div>
    <div
      class="flex items-center"
    />
  </div>
</div>
`;

exports[`Note component > renders with read more link 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  class="text-black bg-success"
  role="alert"
>
  <div
    class="mx-auto flex min-h-[42px] items-center px-5 justify-start sm:min-h-11 sm:pl-7 sm:pr-0 sm:justify-center sm:gap-x-[20%]"
  >
    <div
      class="flex flex-col flex-wrap items-start py-4 sm:flex-row sm:items-center sm:gap-x-5 sm:gap-y-2 sm:py-5"
    >
      <div
        class="font-poppins text-heading-4 sm:text-heading-3"
      >
        Test Title
      </div>
      <div
        class="mt-1 sm:mt-0 text-body-sm"
      >
        Test Description
      </div>
      <a
        class="font-poppins mt-4 text-nowrap rounded-[30px] bg-white px-6 py-[10px] text-button-md hover:underline focus-visible:outline focus-visible:outline-[3px] focus-visible:outline-offset-[1.5px] focus-visible:outline-white active:bg-accent active:text-white active:no-underline"
        href="/"
        role="button"
      >
        Read more
      </a>
    </div>
    <div
      class="flex items-center"
    />
  </div>
</div>
`;

exports[`Note component > renders with success variant 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  class="text-black bg-success"
  role="alert"
>
  <div
    class="mx-auto flex min-h-[42px] items-center px-5 justify-start sm:min-h-11 sm:pl-7 sm:pr-0 sm:justify-center sm:gap-x-[20%]"
  >
    <div
      class="flex flex-col flex-wrap items-start py-4 sm:flex-row sm:items-center sm:gap-x-5 sm:gap-y-2 sm:py-5"
    >
      <div
        class="font-poppins text-heading-4 sm:text-heading-3"
      >
        Test Title
      </div>
      <div
        class="mt-1 sm:mt-0 text-body-sm"
      >
        Test Description
      </div>
    </div>
    <div
      class="flex items-center"
    />
  </div>
</div>
`;

exports[`Note component > renders with warning variant 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  class="text-black bg-warning"
  role="alert"
>
  <div
    class="mx-auto flex min-h-[42px] items-center px-5 justify-start sm:min-h-11 sm:pl-7 sm:pr-0 sm:justify-center sm:gap-x-[20%]"
  >
    <div
      class="flex flex-col flex-wrap items-start py-4 sm:flex-row sm:items-center sm:gap-x-5 sm:gap-y-2 sm:py-5"
    >
      <div
        class="font-poppins text-heading-4 sm:text-heading-3"
      >
        Test Title
      </div>
      <div
        class="mt-1 sm:mt-0 text-body-sm"
      >
        Test Description
      </div>
    </div>
    <div
      class="flex items-center"
    />
  </div>
</div>
`;
